lib/addons/fluxcd/gitrepository.ts

```diff
@@ -0,0 +1,30 @@
+import { FluxCDAddOnProps } from '.';
+import * as spi from "../../spi";
+
+/**
+ * Flux GitRepository API defines a Source to produce an Artifact for a Git repository revision.
+ */
+export class FluxGitRepository {
+
+    constructor(private readonly bootstrapRepo: spi.ApplicationRepository | undefined) {}
```

Comment:
Unclear on what this constructor is for? You are creating a local member variable called `bootstrapRepo` which maybe undefined. You never use then. 

Let's use it and make it mandatory / change the generate method to pass ns and sync interval. This will take care of passing extra props from the addon

Reply:
Fixed.